static/js/admin.js

```diff
@@ -86,6 +92,55 @@ function get_realm_level_notification_settings(options) {
         all_notifications_settings.show_push_notifications_tooltip;
 }
 
+function insert_demo_organization_warning() {
+    const days_remaining = get_demo_organization_deadline_days_remaining();
+    const rendered_demo_organization_warning = render_demo_organization_warning({
+        is_demo_organization: page_params.demo_organization_scheduled_deletion_date,
+        is_owner: page_params.is_owner,
+        days_remaining,
+    });
+    $(".organization-box").find(".settings-section").prepend(rendered_demo_organization_warning);
+}
+
+function handle_demo_organization_conversion() {
+    $(".demo-organization-warning").on("click", () => {
+        const parts = new URL(page_params.realm_uri).hostname.split(".");
+        const subdomain = parts[0];
+        const domain = parts[1].concat(".", parts[2]);
+        const html_body = render_convert_demo_organization_form({
+            realm_subdomain: subdomain,
+            realm_domain: domain,
+            email: page_params.email,
+        });
+
+        function submit_subdomain() {
+            const $string_id = $("#demo-organization-conversion-dialog-widget")
+                .find("#convert-demo-organization-form")
+                .find("input[name='string_id']");
+            const data = {
+                string_id: $string_id.val(),
+            };
+            const opts = {
+                success_continuation(data) {
+                    window.location.href = data.realm_uri;
+                },
+            };
+            const $status_field = $("#demo-organization-conversion-dialog-widget")
+                .find("#dialog_error")
+                .expectOne();
+            settings_ui.do_settings_change(channel.patch, "/json/realm", data, $status_field, opts);
+        }
+
+        dialog_widget.launch({
+            html_heading: $t({defaultMessage: "Make organization permanent"}),
+            html_body,
+            on_click: submit_subdomain,
+            html_submit_button: $t({defaultMessage: "Make organization permanent"}),
```

Comment:
Hm, that action button label seems a bit long/complicated. I think either a simple "Confirm" might be OK, or else "Convert". What do you guys think?

Reply:
I like "Confirm" (see screenshot below) more than "Convert". 

Looking at this, do we want that submit button to be disabled until they update the URL? I assume that no one would intentionally choose their randomly generated demo organization name. Currently, there's an error returned if you submit without changing the URL (see 2nd screenshot below).

<details>
<summary>Confirm button text screenshot</summary>

![Screenshot from 2022-08-23 16-02-27](https://user-images.githubusercontent.com/63245456/186179270-c4c8eb7a-fc1b-48ea-8bb1-4e3065e86d06.png)
</details>

<details>
<summary>Error message when URL isn't changed</summary>

![Screenshot from 2022-08-23 16-04-21](https://user-images.githubusercontent.com/63245456/186179519-70d29462-4d09-444c-9723-754f1b7285a4.png)
</details>

Reply:
With the current updates, the submit button text is "Convert".

Reply:
With the proposal below to leave the input field blank to start with, I don't think checking whether the URL matches the current one will really come up.

Disabling the "Confirm" button until you actually enter something would be nice; we do that in a bunch of places.